### PR TITLE
Remove deprecated Go modules

### PR DESCRIPTION
**Pull request by bkabrda**
### What does this PR do?

Removes modules deprecated by https://github.com/DataDog/datadog-agent/pull/11244

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by albertvaka — APPROVED** [review]
Lgtm as long as tests pass. This is not user-facing so the release note shouldn't be needed, though.

**Comment by bkabrda** [comment]
@albertvaka I think it's user-facing in the sense that in theory somebody might have used these modules from a different Go project, couldn't they?

**Review by djmitche — APPROVED** [review]

File: releasenotes/notes/remove-deprecated-modules-bc2264a08f617ef7.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+upgrade:
+  - |
+    Deprecated modules ``pkg/util/log``, ``pkg/util/scrubber`` and
```
Comment (kayayarai) [inline releasenotes/notes/remove-deprecated-modules-bc2264a08f617ef7.yaml:11]:
```suggestion
    Deprecated modules ``pkg/util/log``, ``pkg/util/scrubber``, and
```

**Review by kayayarai — APPROVED** [review]
Minor edit 👍

**Pull request merged**